lib/std/thread.zig

```diff
@@ -484,6 +484,9 @@ pub const Thread = struct {
             };
             return @intCast(usize, count);
         }
+        if (std.Target.current.os.tag == .haiku) {
```

Comment:
You can get this either via `get_system_info` (a Haiku-specific API), or via `sysconf(_SC_NPROCESSORS_CONF)` (which probably works on other OSes, too.)

Reply:
added